src/main/scala/gameengine/GameEngine.scala

```diff
@@ -0,0 +1,109 @@
+package gameengine
+
+import characters.{GameBoard, Person, Player, Resources}
+
+import scala.Console.println
+import scala.collection.mutable.ListBuffer
+import scala.io.StdIn.readLine
+import scala.util.Random
+
+class GameEngine {
+  val resources = new Resources
+  private var questionList: List[String] = resources.listOfQuestions
+  private val random = new Random()
+  // Instantiate the game board
+  val gameBoard = new GameBoard(resources.charactersList)
+  // create firstPlayer
+  private val firstPlayer: Player = createPlayerAndAssignGameBoard()
+  private var continuedPlaying: Boolean = true
+
+  /**
+   * Create an object of player by assigning name, gameBoard and secret characters
+   * @return player object
+   */
+  def createPlayerAndAssignGameBoard(): Player = {
+    val player = new Player(name = "Player1", gameBoard = gameBoard.gameBoardForPlayer,
+      selectRandomCharacter(resources.charactersList))
+    player
+  }
+
+
+  /**
+   * Select random character by from Character List from Resources class
+   * @param characterList list of characters from resources class
+   * @return randomCharacter of type Person
+   */
+  def selectRandomCharacter(characterList: List[Person]): Person = {
+    val randomCharacter = characterList(random.nextInt(characterList.length))
+    randomCharacter
+  }
+
```

Comment:
Great use of index positions in a list to select a random character. But again no need to assign to a val and return the val

Reply:
We have removed val and return  randomCharacter